### ISSUE TITLE
[Idea]:  Stochastic Gradient Descent (SGD)

### ISSUE THREAD

**Issue by ananyag309** [Ajay-Dhangar/algo#1368]
### Idea Title

Stochastic Gradient Descent (SGD)

### Idea Description

 Stochastic Gradient Descent updates the model's parameters by calculating the gradient of the loss function for a single training example (or a small batch), rather than the entire dataset. This makes it computationally efficient but can lead to noisy updates

### Potential Benefits

- [ ] Improves user experience
- [X] Enhances learning
- [ ] Adds a new feature
- [X] Improves code quality
- [ ] Other
